chips/cc26xx/src/uart.rs

```diff
@@ -0,0 +1,218 @@
+//! UART driver, cc26xx family
+use core::cell::Cell;
+use gpio;
+use ioc;
+use kernel;
+use kernel::common::regs::{ReadOnly, ReadWrite, WriteOnly};
+use kernel::hil::gpio::Pin;
+use kernel::hil::uart;
+use prcm;
+
+pub const UART_BASE: usize = 0x4000_1000;
+pub const MCU_CLOCK: u32 = 48_000_000;
+
+#[repr(C)]
+struct Registers {
+    dr: ReadWrite<u32>,
+    rsr_ecr: ReadWrite<u32>,
+    _reserved0: [u8; 0x10],
+    fr: ReadOnly<u32, Flags::Register>,
+    _reserved1: [u8; 0x8],
+    ibrd: ReadWrite<u32, IntDivisor::Register>,
+    fbrd: ReadWrite<u32, FracDivisor::Register>,
+    lcrh: ReadWrite<u32, LineControl::Register>,
+    ctl: ReadWrite<u32, Control::Register>,
+    ifls: ReadWrite<u32>,
+    imsc: ReadWrite<u32, Interrupts::Register>,
+    ris: ReadOnly<u32, Interrupts::Register>,
+    mis: ReadOnly<u32, Interrupts::Register>,
+    icr: WriteOnly<u32, Interrupts::Register>,
+    dmactl: ReadWrite<u32>,
+}
+
+pub static mut UART0: UART = UART::new();
+
+register_bitfields![
+    u32,
+    Control [
+        UART_ENABLE OFFSET(0) NUMBITS(1) [],
+        TX_ENABLE OFFSET(8) NUMBITS(1) [],
+        RX_ENABLE OFFSET(9) NUMBITS(1) []
+    ],
+    LineControl [
+        FIFO_ENABLE OFFSET(4) NUMBITS(1) [],
+        WORD_LENGTH OFFSET(5) NUMBITS(2) [
+            Len5 = 0x0,
+            Len6 = 0x1,
+            Len7 = 0x2,
+            Len8 = 0x3
+        ]
+    ],
+    IntDivisor [
+        DIVISOR OFFSET(0) NUMBITS(16) []
+    ],
+    FracDivisor [
+        DIVISOR OFFSET(0) NUMBITS(6) []
+    ],
+    Flags [
+        TX_FIFO_FULL OFFSET(5) NUMBITS(1) []
+    ],
+    Interrupts [
+        ALL_INTERRUPTS OFFSET(0) NUMBITS(12) []
+    ]
+];
+
+pub struct UART {
+    regs: *const Registers,
+    client: Cell<Option<&'static uart::Client>>,
+    tx_pin: Cell<Option<u8>>,
+    rx_pin: Cell<Option<u8>>,
+}
+
+impl UART {
+    pub const fn new() -> UART {
+        UART {
+            regs: UART_BASE as *mut Registers,
+            client: Cell::new(None),
+            tx_pin: Cell::new(None),
+            rx_pin: Cell::new(None),
+        }
+    }
+
+    pub fn set_pins(&self, tx_pin: u8, rx_pin: u8) {
+        self.tx_pin.set(Some(tx_pin));
+        self.rx_pin.set(Some(rx_pin));
+    }
+
+    pub fn configure(&self, params: kernel::hil::uart::UARTParams) {
+        let tx_pin = match self.tx_pin.get() {
+            Some(pin) => pin,
+            None => panic!("Tx pin not configured for UART"),
+        };
+
+        let rx_pin = match self.rx_pin.get() {
+            Some(pin) => pin,
+            None => panic!("Rx pin not configured for UART"),
+        };
+
+        unsafe {
+            /*
+             * Make sure the TX pin is output/high before assigning it to UART control
+             * to avoid falling edge glitches
+             */
```

Comment:
Small preference for `//` comments in tree to allow for easier block commenting during random work

Reply:
Sure, I'll fix it!